### DISCUSSION TITLE
Ad Blocker blocking page name analytics

### DISCUSSION THREAD

**Discussion by rohanreddy888** [vercel/next.js#37699] [Help] [answered]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.0.0: Tue May 24 20:30:32 PDT 2022; root:xnu-8792.0.50.111.3~5/RELEASE_ARM64_T8103
    Binaries:
      Node: 16.14.0
      npm: 8.12.1
      Yarn: 1.22.10
      pnpm: N/A
    Relevant packages:
      next: 12.1.7-canary.38
      react: 18.1.0
      react-dom: 18.1.0

### What browser are you using? (if relevant)

Chrome Version 102.0.5005.115 (Official Build) (arm64)

### Describe the Bug

Created a folder in pages called dashboard and created a file called analytics.js and added some content it was not displaying anything on the browser changed the file name and function name from analytic to analytic-one then it was working!

<img width="1440" alt="Screenshot 2022-06-14 at 9 20 38 PM" src="https://user-images.githubusercontent.com/12649422/173621244-e2a41180-3726-4d18-b3bd-adb3aa2e8dcb.png">
<img width="1440" alt="Screenshot 2022-06-14 at 9 20 22 PM" src="https://user-images.githubusercontent.com/12649422/173621239-d25e3fd6-ae60-4b61-b913-8edf77cea5bb.png">[Uploading Screenshot 2022-06-14 at 9.20.12 PM.png…]()

### Expected Behavior

Page not displaying if the file name as analytics

### To Reproduce

Created a folder in pages called dashboard and created a file called analytics.js and added some content it was not displaying anything on the browser changed the file name and function name from analytic to analytic-one then it was working!

**Comment by rohanreddy888** [comment]
<img width="1440" alt="Screenshot 2022-06-14 at 9 20 12 PM" src="https://user-images.githubusercontent.com/12649422/173621671-4f4b49e7-fe3a-4f09-8bc1-44adbc471e9e.png">

**Reply by anch09** [reply]
Hey @rohanreddy888 did you get a solution for this? I'm having the same issue

**Reply by rohanreddy888** [reply]
Yes some urls are not allowed in NextJS you can create a different url it will work.

**Comment by wwwdepot** [comment]
Why don't you open up the web console in chrome and see what's wrong when loading the analytics page.

~I suppose `analytics.js` is a very bad filename. Ad-blockers are very likely going to block it.~ 
Update, analytics page works in a blank new project. In your case, please provide a reproduction example.

**Reply by rohanreddy888** [reply]
Adblocker is the issue it was blocking the content.

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Perhaps you have a browser extension that prevents tracking aggressively? We had issues with an ad blocker that vanished all pages that contained `ad` in the url. I reckon, `analytics` is also part of the url of a very famous tracker... Try browsing in incognito mode, and see if the same happens.

**Reply by rohanreddy888** [reply]
Hi,

Disabled the Adblocker and it's working now. Thank you.

**Reply by icyJoseph** [reply]
Nice! Please consider making as answered by this comment. This issue is fairly difficult to spot!

**Reply by elie222** [reply]
This doesn't actually solve the issue for your users though. So many of them will have ad blockers on and will face the same issues.

**Reply by icyJoseph** [reply]
I think there's a way out using rewrites, so that request arrives with another name to the server. Haven't tested it personally though.

**Comment by elie222** [comment]
Ya, we use this a bit but the point is we don't want the url changed.

I want the url to be the name I choose for users and not have to rename routes (same impact as a redirect).

The annoyance is that it's the preload that crashes the page. Navigating to the page direct doesn't cause an issue (as it's the actual url).

Could also remove preloading from our Next Links but would be nice to still load a page even if the preload has failed.

Best,
Eliezer

**Comment by aa-jeremyhoover** [comment]
Just looking for other solutions for this issue as there currently isn't a great one for users, aside from changing the slug.

Is it not possible to catch this error in an [ErrorBoundary](https://nextjs.org/docs/advanced-features/error-handling) ?

**Reply by rohanreddy888** [reply]
Since it's blocked by the adblocker it's not caught in the ErrorBoundary
